common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -56,24 +72,51 @@ export default class ConfirmationModal extends React.Component {
       }
     }, 1000);
 
-    this.props.wallet.getAddress().then(address => {
-      this.setState({ address });
-    });
+    this._setWalletAddress(this.props.wallet);
   }
 
   componentWillUnmount() {
     clearTimeout(this.readTimer);
   }
 
+  _getStateFromProps(props: Props) {
+    const { rawTransaction, token } = props;
+    const { value, gasPrice, nonce, to } = rawTransaction;
+    let fixedValue;
+    let toAddress;
+
+    if (token) {
+      const tokenData = ERC20.decodeTransfer(rawTransaction.data);
```

Comment:
Is this something used commonly? `ethereumjs-abi` and most native JS functions use `decode` in function names, and I've not seen that $ prefix for functions before. I would be pretty uncertain of what a function was doing without knowing that.